pkg/controller/storagecluster/volumesnapshotterclasses.go

```diff
@@ -0,0 +1,113 @@
+package storagecluster
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	snapapi "github.com/kubernetes-csi/external-snapshotter/v2/pkg/apis/volumesnapshot/v1beta1"
+	ocsv1 "github.com/openshift/ocs-operator/pkg/apis/ocs/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type SnapShotterType string
+
+const (
+	rbdSnapShotter    SnapShotterType = "rbd"
+	cephfsSnapShotter SnapShotterType = "cephfs"
+)
+
+// secret name and namespace for snapshotter class
+const (
+	snapshotterSecretName      = "csi.storage.k8s.io/snapshotter-secret-name"
+	snapshotterSecretNamespace = "csi.storage.k8s.io/snapshotter-secret-namespace"
+	snapshotterNamePrefix      = "snapshotNamePrefix"
+)
+
+var (
+	snapshotterSecretNameMap = map[SnapShotterType]string{
+		rbdSnapShotter:    "rook-csi-rbd-provisioner",
+		cephfsSnapShotter: "csi-cephfs-secret",
+	}
```

Comment:
@aruniiird rook 1.4.1 release is out with cephfs snapshotter, so please accommodate that too in this PR.